### ISSUE TITLE
Exported transaction history (CSV) is incorrect

### ISSUE THREAD

**Issue by nubbel** [Emurgo/yoroi-frontend#2557]
I'm having a hard time making sense of the transaction history exported from the Yoroi dashboard.
Here's an example that shows the exported data is incorrect:
<img width="1164" alt="Screenshot 2021-12-05 at 23 00 01" src="https://user-images.githubusercontent.com/118781/144767502-bfbb9ee1-92ab-4eea-a9ae-587f70d9c45a.png">

The transaction history in the UI shows a deposit, yet the corresponding line in the CSV is a withdrawal. Also note that the net amount is in the fee column, which seems odd as well.

Please let me know how I can help to resolve this issue. Thanks! :)

**Comment by rdlrt** [comment]
This is a withdrawal tx (you may have deregistered your keys) , which _returned_ the 2 ADA deposit fee (minus tx fee that was required to pay for the withdrawal tx). This was deposit Fee paid when delegating (staking) against a pool, being returned to you.

**Comment by nubbel** [comment]
Thanks for your explanation. You are right, I deregistered my key. It’s still a bit unclear to me why this would classify as a withdrawal. Ultimately, funds are moving into my wallet, not away from it. Am I missing something?

The problem is -without further context- this makes the transaction history useless for accounting purposes. It’s impossible to differentiate this type of withdrawal (my balances increases) from an actual withdrawal, e.g. sending someone ADA (my balance decreases).

**Comment by rdlrt** [comment]
I think from Wallet point of view, the only time you deregister the keys is when you withdraw your rewards. Your balance against Wallet account is divided into two buckets:

1. Base Funds : Which you can operate normally (eg: pay 10 ADA out of 100 ADA to someone, or receive 5 ADA against an address, etc)
2. Rewards: Think of it as a read-only bucket filled by protocol , you cannot send funds to it or cannot take funds out partially from this account

The only interaction with Reward bucket above is submitting a withdrawal transaction which _moves all funds from your rewards bucket to Base Funds_, hence - termed as withdrawal of rewards. The wallet makes it possible for you to - optionally - also deregister keys if needed, but the type of action performed is still withdrawal (just that deregister makes your account also return the 2 ADA deposit). I get that it sounds confusing if you associate withdrawal with taking funds out of wallet, but I believe it's where nomen-clature is tied to above concept on protocol itself.

(PS: There are more technical nuances to simplified statement above)

**Comment by nubbel** [comment]
Thanks for explaining the protocol concept, I've been wondering about this and what you say makes a ton of sense.

The nomenclature is actually not very important to me. What is important is that the transaction history should allow me to reconstruct what happened and in its current form it doesn't:

<img width="1181" alt="Screenshot 2021-12-06 at 20 59 06" src="https://user-images.githubusercontent.com/118781/144923385-bd9f1444-7249-4be0-8040-19d8cb222369.png">

In the screenshot above there are 3 transactions, all of which are `Withdrawal`s in the CSV export. However, in the Yoroi UI the first and the last show up with a negative amount, the middle one with a positive amount.
Said differently, I wouldn't be able to rebuild the UI purely from the information in the CSV. Nor would I be able to hand the CSV to an accountant to do my tax report.

Ideally, there would be entries for both buckets in the CSV, i.e. the `Withdrawal` from the rewards bucket would be matched by a `Deposit` to the base bucket.

I understand that this would be a bit of work, I'd be happy with just the "raw" data, i.e. whatever data is driving the UI.

**Comment by rdlrt** [comment]
I fully agree that the CSV does not provide accurate information in the sense, and furthermore - staking rewards become more complicated depending on region 👍🏻 
@vsubhuman - for your review, suggestions:
1. the "Type" for CSV could include `Intrawallet` flag so that the CSV could make more sense?
2. keyDeposits (deposit + withdrawal) and MIRs could reflect accurately in default comments.

**Comment by jarstorm** [comment]
Hello, and thank you for your contribution.
We're currently reviewing long-standing issues to keep the tracker manageable. As this issue hasn't seen activity in some time, we're closing it.
If the problem persists or you'd like to follow up, feel free to comment and we'll reopen it.
